apps/cookbook/src/app/examples/modal-example/alert-example/modal-example-alert-with-guard-stepper.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'cookbook-modal-example-alert-with-guard-stepper',
+  template: `
+    <ol>
+      <li
+        *ngFor="let step of steps; let i = index"
+        [class.active]="currentStep - 1 === i && !step.steps"
+      >
+        <a
+          *ngIf="i === currentStep - 1 && !!step.route"
```

Comment:
any reason we are not using the new *ngIf that came with angular version 17?
https://blog.angular-university.io/angular-if/

Reply:
@Fuzzy3 I would say code consistency being the main reason. If we want to use it we should apply across the code base as a general refactoring in it's own PR to avoid confusing developers seeing different syntax for the same thing 😊